docs/freqai-feature-engineering.md

```diff
@@ -0,0 +1,165 @@
+            informative_shift = informative_shift.add_suffix("_shift-" + str(n))
+            informative = pd.concat((informative, informative_shift), axis=1)
+
+        df = merge_informative_pair(df, informative, self.config["timeframe"], tf, ffill=True)
+        skip_columns = [
+            (s + "_" + tf) for s in ["date", "open", "high", "low", "close", "volume"]
+        ]
+        df = df.drop(columns=skip_columns)
+
+        # Add generalized indicators here (because in live, it will call this
+        # function to populate indicators during training). Notice how we ensure not to
+        # add them multiple times
+        if set_generalized_indicators:
+            df["%-day_of_week"] = (df["date"].dt.dayofweek + 1) / 7
+            df["%-hour_of_day"] = (df["date"].dt.hour + 1) / 25
+
+            # user adds targets here by prepending them with &- (see convention below)
+            # If user wishes to use multiple targets, a multioutput prediction model
+            # needs to be used such as templates/CatboostPredictionMultiModel.py
+            df["&-s_close"] = (
+                df["close"]
+                .shift(-self.freqai_info["feature_parameters"]["label_period_candles"])
+                .rolling(self.freqai_info["feature_parameters"]["label_period_candles"])
+                .mean()
+                / df["close"]
+                - 1
+            )
+
+        return df
+```
+
+In the presented example, the user does not wish to pass the `bb_lowerband` as a feature to the model,
+and has therefore not prepended it with `%`. The user does, however, wish to pass `bb_width` to the
+model for training/prediction and has therefore prepended it with `%`.
+
+After having defined the `base features`, the next step is to expand upon them using the powerful `feature_parameters` in the configuration file:
+
+```json
+    "freqai": {
+        ...
+        "feature_parameters" : {
+            "include_timeframes": ["5m","15m","4h"],
+            "include_corr_pairlist": [
+                "ETH/USD",
+                "LINK/USD",
+                "BNB/USD"
+            ],
+            "label_period_candles": 24,
+            "include_shifted_candles": 2,
+            "indicator_periods_candles": [10, 20]
+        },
+        ...
+    }
+```
+
+The `include_timeframes` in the config above are the timeframes (`tf`) of each call to `populate_any_indicators()` in the strategy. In the presented case, the user is asking for the `5m`, `15m`, and `4h` timeframes of the `rsi`, `mfi`, `roc`, and `bb_width` to be included in the feature set.
+
+The user can ask for each of the defined features to be included also for informative pairs using the `include_corr_pairlist`. This means that the feature set will include all the features from `populate_any_indicators` on all the `include_timeframes` for each of the correlated pairs defined in the config (`ETH/USD`, `LINK/USD`, and `BNB/USD` in the presented example).
```

Comment:
Just a general comment for the future, not necessarily for this PR:

Throughout the docs, we try to adress the user as the reader (i.e. "you/your"). 

i find the "the user" terminology odd / missplaced in docs - as it assumes that the reader is not necessarily the user. This may sound correct for papers (where you explain to some reader how a user would interact with this) - but is wrong for documentation.

Reply:
I both agree and disagree with this. Using "you/your" makes the docs more personal which is nice when you read them as a user, but not only users will read the docs and for non-users "you/your" becomes weird. Personally, I prefer to keep it as it is as it sounds more professional and generalized.

Reply:
It's fine for now - but for the future, i'd appreciate if we can align the freqAI docs to the remaining docs, using you/your.
This is a project written by volunteers - for people using it for free. 

It intentionally shouldn't sound too professional - and docs that use "the user" always feel like "the write though he is something better, not addressing the reader directly".

Reply:
It's a very fair point and I fully agree that we should not (seemingly) be talking over peoples heads. I have changed "the user/they/their" to "you/your" throughout except for when referring to examples were a "fictive user" is doing things. If I missed anything let me know and I'll fix it in the next commit.

Reply:
Ok, let's try this method of avoiding you/user as much as possible.